### PR TITLE
RavenDB-4902 Review fixes

### PR DESCRIPTION
**Pull request by fitzchak**

### CODE REVIEW

File: src/Raven.Client/Http/ApiKeyAuthenticator.cs [hunk 1]
```diff
@@ -28,7 +28,7 @@ public class ApiKeyAuthenticator : IDisposable
             {
                 try
                 {
-                    Logger.Info("Trying to connect using WebSocket to {0} for authentication", uri);
+                    Logger.Info($"Trying to connect using WebSocket to {uri} for authentication");
```
Comment (ayende) [inline src/Raven.Client/Http/ApiKeyAuthenticator.cs]: if(Logger.IsInfoEnabled)

File: src/Raven.Client/Http/ApiKeyAuthenticator.cs [hunk 2]
```diff
@@ -75,15 +75,14 @@ public class ApiKeyAuthenticator : IDisposable
...
-                        
+                        Logger.Info($"Failed to close the client", ex);
                     }
 
                     return currentToken;
                 }
                 catch (Exception ex)
                 {
-                    Logger.DebugException($"Failed to DoOAuthRequest to {url} with {apiKey}", ex);
+                    Logger.Operations($"Failed to DoOAuthRequest to {url} with {apiKey}", ex);
```
Comment (ayende) [inline src/Raven.Client/Http/ApiKeyAuthenticator.cs]: This isn't an oeprational issue.
We throw an exception upward again

File: src/Raven.Client/Http/ApiKeyAuthenticator.cs [hunk 3]
```diff
@@ -199,7 +198,7 @@ private string[] ExtractApiKeyAndSecret(string apiKey)
             catch (WebSocketException ex)
             {
                 builder?.Dispose();
-                Logger.DebugException("Failed to receive a message, client was probably disconnected", ex);
+                Logger.Info("Failed to receive a message, client was probably disconnected", ex);
```
Comment (ayende) [inline src/Raven.Client/Http/ApiKeyAuthenticator.cs]: IsInfoEnabled

File: src/Raven.Client/Http/RequestExecuter.cs [hunk 4]
```diff
@@ -53,8 +53,9 @@ public RequestExecuter(DocumentStore store)
                 {
                     Database = _store.DefaultDatabase,
                     ApiKey = _store.ApiKey,
-                    Url = _store.Url
-                }
+                    Url = _store.Url,
+                },
+                Etag = int.MinValue
```
Comment (ayende) [inline src/Raven.Client/Http/RequestExecuter.cs]: There isn't a match here!

File: src/Raven.Client/Http/ApiKeyAuthenticator.cs (hunk 1 above)
Comment (fitzchak) [inline src/Raven.Client/Http/ApiKeyAuthenticator.cs]: Fixed

File: src/Raven.Client/Http/RequestExecuter.cs [hunk 5]
```diff
@@ -314,21 +329,38 @@ private ServerNode GetFailoverNode<T>(RavenCommand<T> command)
...
+
+        private void UpdateCurrentTokenCallback(object _)
+        {
+            var topology = _topology;
+
+            var leaderNode = topology.LeaderNode;
+            if (leaderNode != null)
+            {
+#pragma warning disable 4014
+                HandleUnauthorized(null, leaderNode, _context);
```
Comment (ayende) [inline src/Raven.Client/Http/RequestExecuter.cs]: Why are you calling this?

File: src/Raven.Client/Http/ApiKeyAuthenticator.cs (hunk 2 above)
Comment (fitzchak) [inline src/Raven.Client/Http/ApiKeyAuthenticator.cs]: You're right

File: src/Raven.Client/Http/RequestExecuter.cs (hunk 5 above)
Comment (ayende) [inline src/Raven.Client/Http/RequestExecuter.cs:355]: Who is handling errors?

File: src/Raven.Client/Http/RequestExecuter.cs [hunk 6]
```diff
@@ -314,21 +329,38 @@ private ServerNode GetFailoverNode<T>(RavenCommand<T> command)
...
-
             if (string.IsNullOrEmpty(oauthSource))
                 oauthSource = node.Url + "/OAuth/API-Key";
 
             var currentToken = await _authenticator.AuthenticateAsync(oauthSource, node.ApiKey, context).ConfigureAwait(false);
             node.CurrentToken = currentToken;
+
+            if (_updateCurrentTokenTimer == null)
+            {
+                _updateCurrentTokenTimer = new Timer(UpdateCurrentTokenCallback, null, TimeSpan.FromMinutes(20), TimeSpan.FromMinutes(20));
```
Comment (ayende) [inline src/Raven.Client/Http/RequestExecuter.cs:343]: Start this immeidately

File: src/Raven.Client/Http/RequestExecuter.cs [hunk 7]
```diff
@@ -314,21 +329,38 @@ private ServerNode GetFailoverNode<T>(RavenCommand<T> command)
...
+            {
+#pragma warning disable 4014
+                HandleUnauthorized(null, leaderNode, _context);
+#pragma warning restore 4014
+            }
+
+            foreach (var node in topology.Nodes)
+            {
+#pragma warning disable 4014
+                HandleUnauthorized(null, node, _context);
```
Comment (ayende) [inline src/Raven.Client/Http/RequestExecuter.cs:362]: Error handling.

**Comment by fitzchak** [comment]
Fixed, I amended last commit

**Pull request merged**